peps/pep-0747.rst

```diff
@@ -532,6 +530,35 @@ of ``type``.
 .. _designed: https://mail.python.org/archives/list/typing-sig@python.org/message/D5FHORQVPHX3BHUDGF3A3TBZURBXLPHD/
 
 
+Treat ``type[T]`` as a subtype of ``TypeForm[T]``
+-------------------------------------------------
+
+It was suggested that type ``type[T]`` should be considered a subtype
+of ``TypeForm[T]``. This was ultimately rejected because there are ways to
+create an object of type ``type[T]`` that does not encode a valid type expression.
+For example, the expression ``type[1]`` is not a valid type expression. Likewise,
+the  expression ``type[S]`` is not a valid type expression if ``S`` is an
+out-of-scope type variable. This same argument applies to other special forms
+like objects of type ``UnionType``, which may or may not encode a valid type
+expression. Rather than carving out a special case for ``type[T]`` that would
+allow for potential unsoundness, it was decided to treat all type forms
```

Comment:
I also don't think there is any possible unsoundness that could result from making `type[T]` a subtype of `TypeForm[T]`, because the runtime objects that inhabit `type[T]` (the Python type/class object `T` and its subclasses) must already be handled by any function taking `TypeForm[T]`, since a type/class object arising from a valid type expression and one arising from an invalid type expression are indistinguishable at runtime. The argument for making `type[T]` not a subtype of `TypeForm[T]` is a purely theoretical one, deriving from the requirement that only a valid type expression can evaluate to a `TypeForm[T]` type.